store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -9451,6 +9483,45 @@ public long countObjects(CountObjectsType type, Domain domain, UCService ucServi
         return num;
     }
 
+    @Override
+    public long countObjects(CountObjectsType type, Object... args) throws ServiceException {
+        ZLdapFilter filter = null;
+
+        // setup types for finding bases
+        Set<ObjectType> types = Sets.newHashSet();
+        int argsLength = args.length;
+        String ldapAttribute = null;
+        switch (type) {
+            case internalUserAccountsByCosesWithLdapFeature: // multiple cos ids
+                List<String> cosIds = (List<String>) args[0];
+                ldapAttribute = (String) args[1];
+                types.add(ObjectType.accounts);
+                filter = filterFactory.accountsByCosesAndFeatureCheck(cosIds, ldapAttribute);
+                break;
+            case internalUserAccountsByCosWithLdapFeature:// single cos id
+                String cosId = (String) args[0];
+                ldapAttribute = (String) args[1];
+                types.add(ObjectType.accounts);
+                filter = filterFactory.accountsByCosAndFeatureCheck(cosId, ldapAttribute);
+                break;
+            case internalUserAccountsWithLdapFeatureCheck: // accounts with a feature enabled or disabled
+                ldapAttribute = (String) args[0];
+                String ldapValue = (String) args[1];
+                types.add(ObjectType.accounts);
```

Comment:
Avoid redundancy in the code by moving out `ObjectType.accounts` to `types` set, out of the `switch` block.

Reply:
here we cannot move this out of the switch block, as it can vary on new upcoming cases. for now whatever the cases we have all are using same type. but there is a possibility of changing it on new case.